include/seqan3/io/stream/parse_condition.hpp

```diff
@@ -0,0 +1,468 @@
+    bool operator()(value_t && val) const
+    {
+        using seqan3::to_char;
+        using seqan3::assign_char;
+
+        if constexpr (std::is_same_v<std::decay_t<value_t>, std::decay_t<alphabet_t>>)
+            return true;
+
+        alphabet_t cmp;
+        assign_char(cmp, to_char(val));
+        return to_char(cmp) == std::toupper(static_cast<uint8_t>(to_char(val)));
+    }
+};
+
+// ----------------------------------------------------------------------------
+// is_char
+// ----------------------------------------------------------------------------
+
+/*!\brief Parse condition that checks if a given value is equal to `char_v`.
+ * \implements seqan3::detail::parse_condition_concept
+ * \tparam alphabet_t non-type template parameter with the value that should be checked against.
+ * \ingroup io
+ */
+template <char char_v>
+struct is_char : public parse_condition<is_char<char_v>>
+{
+    //!\brief The message representing this condition.
+    static constexpr auto msg = constexpr_string{"is_char<"} +
+                                constexpr_string{char_v} +
+                                constexpr_string(">");
+
+    //!\brief Invokes the condition check for `val`. `val` must satisfy the seqan3::alphabet_concept.
+    template <alphabet_concept value_t>
+    bool operator()(value_t && val) const
+    {
+        return to_char(val) == char_v;
+    }
+};
+
+} // namespace seqan3::detail
+
+namespace seqan3
+{
+
+// ----------------------------------------------------------------------------
+// Composite Functors
+// ----------------------------------------------------------------------------
+// Don't use isblank() or isspace() as it they seem to be slower than our functors (due to inlining)
+
+/*\brief A view that calls seqan3::to_char() on each element in the input range.
+ * \tparam irng_t The type of the range being processed. See below for requirements. [template parameter is omitted in pipe notation]
+ * \param irange The range being processed. [parameter is omitted in pipe notation]
+ * \returns A range of converted elements. See below for the properties of the returned range.
+ * \ingroup io
+ *
+ * ### View properties
+ *
+ * | Composite Functor   | Tested Condition                            |
+ * |---------------------|---------------------------------------------|
+ * | is_tab              | v = '\\t'                                   |
+ * | is_space            | v = ' '                                     |
+ * | is_blank            | v = '\\t' \|\| ' '                          |
+ * | is_new_line         | v = '\\n' \|\| '\\r'                        |
+ * | is_whitespace       | v = ' ' \|\| '\\t' \|\| '\\n' \|\| '\\r'    |
+ * | is_alpha            | v = 'a'...'z' \|\| 'A'...'Z'                |
+ * | is_digit            | v = '0'...'9'                               |
+ * | is_alpha_num        | v = 'a'...'z' \|\| 'A'...'Z' \|\| '0'...'9' |
+ * | is_graph            | v = '!'...'~'                               |
```

Comment:
**_Graph_** may be misleading. Maybe find another word, as you mean any printable symbol/non-whitespace. 
Suggestion: `is_symbol`